api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -806,7 +806,14 @@ public String uploadMappings(MultipartFile mapping) {
 		String mappingFolderName = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_MAPPINGS_FOLDER
 		        + File.separator;
 		String mappingName = mapping.getOriginalFilename();
-		
+
+		boolean isMappingsFolderCreated = (new File(mappingFolderName)).mkdirs();
+
+		if (!isMappingsFolderCreated) {
+			msg = "Something went wrong while creating the mappings folder";
```

Comment:
thanks @Gravewalker666  why not put this error message under the messages.properies file

Reply:
Sure, will do

Reply:
Resolved with the commit b02f4aa
Please review @HerbertYiga @k-joseph

Reply:
@Gravewalker666  have you tested this with a running instance of openmrs?

Reply:
Yes, I changed a code a bit, I think this is the optimum way of checking for the folder 
@HerbertYiga can you please review the code again??